### DISCUSSION TITLE
Mqtt publish but only "locally"

### DISCUSSION THREAD

**Discussion by MatteoRampazzo** [cesanta/mongoose#2730] [Ask questions, get answers] [answered]
Hi, welcome to everyone.
I'm trying to develop a telemetry solution using Mongoose based on a ST custom board based on a F446 with the ethernet module W5500.
Studying your documentation I've been able to configure the W5500 driver using SPI functions provided by ST, and all works quite well.
All the simple example with http dashboard work, also the MQTT example, but here start the problem.
When I publish an MQTT message, sometimes it reach the broker and other times no, I've first thought that was related to the too high frequency (I've run some test trying to publish 100 or 1000 message per seconds) but it happens also at low frequency.

I try to explain myself better, debugging over UART or the IDE debugger I see every message being published, but watching the broker not every one arrive, it's not a intermittent problem.
They can arrive for 1 minute after 2/3 minute that the board power on, after  they don't arrive for other 5 minute, and maybe restart to arrive.
I've tried to change the timeout of the poll function and other variables but not seeing any relevant difference.

<img width="677" alt="Screenshot 2024-04-26 alle 19 39 43" src="https://github.com/cesanta/mongoose/assets/43091722/e823430b-8433-4cf6-9392-966c03dd71a9">
<img width="677" alt="Screenshot 2024-04-26 alle 19 39 30" src="https://github.com/cesanta/mongoose/assets/43091722/779f3400-b936-4586-8244-3ffd03fb7e1d">
<img width="677" alt="Screenshot 2024-04-26 alle 19 39 21" src="https://github.com/cesanta/mongoose/assets/43091722/eeb35cbb-621f-4fd2-89d2-93fb0917e6e3">
<img width="677" alt="Screenshot 2024-04-26 alle 19 38 54" src="https://github.com/cesanta/mongoose/assets/43091722/39dbd84e-58f2-4cf6-a8e6-bac8bb5cc9f7">

<img width="677" alt="Screenshot 2024-04-26 alle 19 43 36" src="https://github.com/cesanta/mongoose/assets/43091722/97392d5a-671c-4925-8b6c-19bf05c14a16">
<img width="677" alt="Screenshot 2024-04-26 alle 19 43 47" src="https://github.com/cesanta/mongoose/assets/43091722/c08c3371-fdf1-4f70-a716-17c6a2578af4">

**Comment by scaprile** [comment, marked as answer]
I'm sorry, there is no such thing as "sometimes reaches sometimes not". Your Ethernet frames either go out or don't, and they are fine or contain garbage.  Your broker then either receives the good frames or it does not. Please sniff your network (using Wireshark, for example) and observe the traffic.
There is no "too high frequency", what you are doing is sending at some interval, every time you call the 'pub' function that will place a message on the TCP buffer that will be sent by the manager, if you try to send too much, you will get out of memory and your log will indicate that.
You may have hardware issues, use an oscilloscope or logic analyzer; the HAL SPI driver might be being used improperly or be buggy, your broker might be discarding messages, QoS0 msgs are not guaranteed to be delivered, your network might have issues, etc.

Please do not paste code snapshots, use markdown or attach.
I don't have the slightest clue of what are those white bars on a black background and what are you trying to mean with that. The proof that an Ethernet frame went out is seeing it on a sniffer capture, and if it is not out, then you have to check the SPI comm with the W5500.

If, after your network and hardware inspection, you find an actual issue, feel free to open an issue, fill the issue template, and provide us with pertinent information (including useful network captures) in order to reproduce and fix the issue. We don't inspect your code for errors, please send the minimum piece of code that can reproduce the issue
